### PR TITLE
rewrite module with native fact

### PR DESCRIPTION
**Pull request by b4ldr**
This is a bit of a rewrite however it seems like it also simplifies
things somewhat.  I think the end result should be the same result but
hard to test.

i havealso  removed the apt::source stuff as lldpd is in currnt versions of debian.
As a side comment i'm not a fan of having a manage_repo flag in a
module.  i think such things should be handled at the  profile/role level
and not in a module like this but note sure what other views are?

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -29,14 +29,12 @@
     {
       "operatingsystem": "CentOS",
       "operatingsystemrelease": [
-        "6",
```
Comment (bastelfreak) [inline metadata.json:32]: to have a prettier changelog, we prefer one change in one PR. can you submit one that drops support for CentOS/RHEL/VZ 6 and one for Debian 8?

File: metadata.json [hunk 2]
```diff
@@ -49,15 +47,13 @@
     {
       "operatingsystem": "Ubuntu",
       "operatingsystemrelease": [
-        "16.04",
```
Comment (bastelfreak) [inline metadata.json:52]: I think Ubuntu 16.04 is still supported and works? In that case I think we can keep it?

File: lib/facter/lldb.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require 'facter'
+require 'json'
+
+Facter.add(:lldp) do
+  confine do
+    Facter::Util::Resolution.which('lldpctl')
+  end
+
+  setcode do
+    data = Facter::Util::Resolution.exec('lldpctl -f json')
```
Comment (bastelfreak) [inline lib/facter/lldb.rb:10]: I implemented this as an external fact because I wasn't sure how long lldpctl takes to run / if it would block the puppet run. But getting rid of the systemd timer/jq dependency is also nice. Did you do some testing?

File: lib/facter/lldb.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    json = JSON.parse(data)
```
Comment (bastelfreak) [inline lib/facter/lldb.rb:11]: I had some strange edge cases where lldpctl didn't produce any output / printend warnings before the actual json. that's why the script valides the output with jq.

File: metadata.json (hunk 1 above)
Comment (b4ldr) [reply metadata.json:32]: ack ill split theses up later tonight

File: metadata.json (hunk 2 above)
Comment (b4ldr) [reply metadata.json:52]: ack will add back

File: lib/facter/lldb.rb (hunk 3 above)
Comment (b4ldr) [reply lib/facter/lldb.rb:10]: Just did a test on a vm hosts which has 23 neighbours and runs much pretty fast

```shell
$ for i in {1..10} ; do time sudo  lldpctl -f json > /dev/null ; done                                               
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 88% cpu 0.021 total
sudo lldpctl -f json > /dev/null  0.00s user 0.02s system 88% cpu 0.023 total
sudo lldpctl -f json > /dev/null  0.01s user 0.00s system 86% cpu 0.020 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 86% cpu 0.024 total
sudo lldpctl -f json > /dev/null  0.00s user 0.01s system 88% cpu 0.019 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 90% cpu 0.020 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 86% cpu 0.020 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 85% cpu 0.016 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 86% cpu 0.019 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 89% cpu 0.021 total
```

File: lib/facter/lldb.rb (hunk 4 above)
Comment (b4ldr) [reply lib/facter/lldb.rb:11]: I did wonder why you used jq, could add a try; catch here.  Guessing you don't remember the edge cases or have test data to tickle the bug?

As mentioned above we do make use of this binary to [produce a similar fact](https://github.com/wikimedia/puppet/blob/production/modules/base/lib/facter/lldp.rb) and don't see any issues with.  however we use `-f xml` so depending on where the bug is might not hit it

Comment (b4ldr) [reply lib/facter/lldb.rb]: added a rescue

File: data/VirtuozzoLinux.yaml [hunk 5]
```diff
@@ -1,2 +1,2 @@
----
+lldpd::manage_repo: true
```
Comment (bastelfreak) [inline data/VirtuozzoLinux.yaml:1]: can you please keep the `---` in the first line?

File: manifests/repo.pp [hunk 6]
```diff
@@ -1,34 +0,0 @@
...
-      'RedHat': {
-        yumrepo { 'lldpd':
-          ensure  => 'present',
-          descr   => 'lldpd toolset',
-          gpgkey  => "https://download.opensuse.org/repositories/home:/vbernat/${lldpd::repourl}/repodata/repomd.xml.key",
-          baseurl => "https://download.opensuse.org/repositories/home:/vbernat/${lldpd::repourl}/",
-        }
-      }
-      'Debian': {
-        apt::source { 'lldpd':
```
Comment (bastelfreak) [inline manifests/repo.pp:21]: did this get lost or did you purge it on purpose?

Comment (b4ldr) [reply manifests/repo.pp:21]: As far as i cant tell lldpd is available in stock stretch and buster.  I assumed it would be in the ubuntu varients but i havn't checked

File: data/VirtuozzoLinux.yaml (hunk 5 above)
Comment (b4ldr) [reply data/VirtuozzoLinux.yaml:1]: ahh yes will add back

File: manifests/repo.pp (hunk 6 above)
Comment (bastelfreak) [reply manifests/repo.pp:21]: I added those repos because the debian repos contain pretty old lldpd packages :(

Comment (b4ldr) [reply manifests/repo.pp:21]: Added back and reset the defaults.

**Comment by b4ldr** [comment]
I accidently pushed an early [commit](https://github.com/voxpupuli/puppet-lldpd/commit/11023cb55f3cfd86fd80d461448595e1f874491e) which re-wrote the previous modsync 4 commit to include the changes in this branch. Pretty sure i have fixed it now but let me know if you want mt to create a clean PR

**Comment by b4ldr** [comment]
The failures now seem more general and related to beaker install_puppet

**Comment by b4ldr** [comment]
even though i expect it to be compatible and not really breaking i feel like we should mark this as a breaking change and do a major release as its a significant change to the implmentations and dosn;t clean up the old files

**Review by bastelfreak — APPROVED** [review]

**Comment by bastelfreak** [comment]
@b4ldr the code looks fine. but should we implement a cleanup to purge the old cronjobs/systemd files?

**Review by bastelfreak — APPROVED** [review]

**Review by bastelfreak — APPROVED** [review]
